xds/internal/server/rds_handler.go

```diff
@@ -131,32 +139,58 @@ func (rh *rdsHandler) close() {
 	}
 }
 
+type rdsWatcherUpdate struct {
+	update *xdsresource.RouteConfigUpdate
+	err    error
+}
+
 // rdsWatcher implements the xdsresource.RouteConfigWatcher interface and is
 // passed to the WatchRouteConfig API.
 type rdsWatcher struct {
 	parent    *rdsHandler
 	logger    *igrpclog.PrefixLogger
 	routeName string
+
+	mu        sync.Mutex
+	cancelled bool // eats callbacks if true
```

Comment:
IDK how much I care but note the Go prefers the "American" spelling which is "canceled".  Hence we have `context.Canceled` and `codes.Canceled` (even though the Google RPC status codes proto has `CANCELLED`).  So it's probably best to be consistent, but pretty minor since it's internal.

Reply:
Ah, ok noted for future haha. Switched.